### ISSUE TITLE
Failure with NPM and stack overflow while building this action image

### ISSUE THREAD

**Issue by sixlettervariables** [thollander/actions-comment-pull-request#17]
[Today we started getting the following error during our workflows](https://github.com/MegaMek/mekhq/pull/2156/checks?check_run_id=1294417859), not sure if this is an `npm` issue or the `package-lock.json` in this repo:
```
Build container for action use: '/home/runner/work/_actions/thollander/actions-comment-pull-request/master/Dockerfile'.
  /usr/bin/docker build -t 1e5c35:ad639c88d707402b8e5ba2e8e25386b6 -f "/home/runner/work/_actions/thollander/actions-comment-pull-request/master/Dockerfile" "/home/runner/work/_actions/thollander/actions-comment-pull-request/master"
  Sending build context to Docker daemon  36.35kB
  
  Step 1/4 : FROM node:slim
  slim: Pulling from library/node
  babf97a3f00a: Pulling fs layer
  d81eb1809d95: Pulling fs layer
  54b0718e1fd5: Pulling fs layer
  9f80bf52a6c0: Pulling fs layer
  28724063fd29: Pulling fs layer
  9f80bf52a6c0: Waiting
  28724063fd29: Waiting
  d81eb1809d95: Verifying Checksum
  d81eb1809d95: Download complete
  9f80bf52a6c0: Verifying Checksum
  9f80bf52a6c0: Download complete
  babf97a3f00a: Verifying Checksum
  babf97a3f00a: Download complete
  28724063fd29: Verifying Checksum
  28724063fd29: Download complete
  54b0718e1fd5: Verifying Checksum
  54b0718e1fd5: Download complete
  babf97a3f00a: Pull complete
  d81eb1809d95: Pull complete
  54b0718e1fd5: Pull complete
  9f80bf52a6c0: Pull complete
  28724063fd29: Pull complete
  Digest: sha256:a39f2854f7fc02c8636c11227ca2d6680bd46a2173168650928b69cc03dffb1c
  Status: Downloaded newer image for node:slim
   ---> 2870b8b46426
  Step 2/4 : COPY . .
   ---> e1de5efdbd73
  Step 3/4 : RUN npm install --production
   ---> Running in 5263be933aac
  npm notice 
  npm notice New patch version of npm available! 7.0.2 -> 7.0.3
  npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.0.3>
  npm notice Run `npm install -g npm@7.0.3` to update!
  npm notice 
  npm ERR! Maximum call stack size exceeded
  
  npm ERR! A complete log of this run can be found in:
  npm ERR!     /root/.npm/_logs/2020-10-22T19_01_38_190Z-debug.log
  The command '/bin/sh -c npm install --production' returned a non-zero code: 1
  
  Warning: Docker build failed with exit code 1, back off 2.14 seconds before retry.
  /usr/bin/docker build -t 1e5c35:ad639c88d707402b8e5ba2e8e25386b6 -f "/home/runner/work/_actions/thollander/actions-comment-pull-request/master/Dockerfile" "/home/runner/work/_actions/thollander/actions-comment-pull-request/master"
  Sending build context to Docker daemon  36.35kB
  
  Step 1/4 : FROM node:slim
   ---> 2870b8b46426
  Step 2/4 : COPY . .
   ---> Using cache
   ---> e1de5efdbd73
  Step 3/4 : RUN npm install --production
   ---> Running in 8e917ec8efea
  npm notice 
  npm notice New patch version of npm available! 7.0.2 -> 7.0.3
  npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.0.3>
  npm notice Run `npm install -g npm@7.0.3` to update!
  npm notice 
  npm ERR! Maximum call stack size exceeded
  
  npm ERR! A complete log of this run can be found in:
  npm ERR!     /root/.npm/_logs/2020-10-22T19_01_41_950Z-debug.log
  The command '/bin/sh -c npm install --production' returned a non-zero code: 1
  
  Warning: Docker build failed with exit code 1, back off 7.085 seconds before retry.
  /usr/bin/docker build -t 1e5c35:ad639c88d707402b8e5ba2e8e25386b6 -f "/home/runner/work/_actions/thollander/actions-comment-pull-request/master/Dockerfile" "/home/runner/work/_actions/thollander/actions-comment-pull-request/master"
  Sending build context to Docker daemon  36.35kB
  
  Step 1/4 : FROM node:slim
   ---> 2870b8b46426
  Step 2/4 : COPY . .
   ---> Using cache
   ---> e1de5efdbd73
  Step 3/4 : RUN npm install --production
   ---> Running in 9362d4ee85ea
  npm notice 
  npm notice New patch version of npm available! 7.0.2 -> 7.0.3
  npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.0.3>
  npm notice Run `npm install -g npm@7.0.3` to update!
  npm notice 
  npm ERR! Maximum call stack size exceeded
  
  npm ERR! A complete log of this run can be found in:
  npm ERR!     /root/.npm/_logs/2020-10-22T19_01_50_702Z-debug.log
  The command '/bin/sh -c npm install --production' returned a non-zero code: 1
  
Error: Docker build failed with exit code 1
```

**Comment by provtimofei** [comment]
Same issue

**Comment by isyutaro** [comment]
is the node version. Try change Dockerfile and use node-14

```
FROM node:14-slim
```

**Comment by ysaunier** [comment]
Same here.

**Comment by fogfish** [comment]
are there any ETA where this issue is resolved ;-) ?

**Comment by danielhalldin** [comment]
Same here.

**Comment by vinod-sai** [comment]
Same here.

**Comment by vinodsai-a** [comment]
From @isyutaro comment i updated the docker file, its working fine. Please accept the pull request @thollander 
This is the pull request id #18
till this is merged, you can use the patch that i created in your workflow 
instead of 
`uses: thollander/actions-comment-pull-request@master` 
please change it to 
`uses: vinodsai-a/actions-comment-pull-request@master` 

@danielhalldin @fogfish @ysaunier @provtimofei @sixlettervariables

**Comment by ertrzyiks** [comment]
The [disclaimer](https://github.com/thollander/actions-comment-pull-request#disclaimer) from the readme also suggest a good alternative of using `actions/github-script` instead of this action.

**Comment by thollander** [comment]
Should be fixed by #18

**Comment by eddiemonge** [comment]
@thollander can you release a new version as well please?
